pvlib/tools.py

```diff
@@ -230,11 +251,16 @@ def _doy_to_datetimeindex(doy, epoch_year=2014):
 
 
 def _datetimelike_scalar_to_doy(time):
-    return pd.DatetimeIndex([pd.Timestamp(time)]).dayofyear
+    return _datetimelike_scalar_to_datetimeindex(time).dayofyear
 
 
 def _datetimelike_scalar_to_datetimeindex(time):
-    return pd.DatetimeIndex([pd.Timestamp(time)])
+
+    timestamp = pd.Timestamp(time)
+
+    timestamp_utc = _pandas_to_utc(timestamp)
```

Comment:
I'm not comfortable converting the timezone in this utility function. I can see the intent, but I think this will be a surprise if this function gets used for something else.

The `pyephem` package handles timezones internally. The NREL method has code that converts to UTC if needed. So these options are timezone-aware. Neither of these algorithms relies on day of year.

What breaks if we don't convert to UTC here?

Reply:
I have agreed and reverted `_datetimelike_scalar_to_datetimeindex` in defddfd . Yet I copied the same conversion logic into both of `solarposition.pyephem_earthsun_distance` and `nrel_earthsun_distance` for the following reasons.

For `pyephem` dni_extra, it seems true that the function is tz-aware but it is also conventional to create a `ephem.Date` with UTC datetimes only as in fa62003 . Please see also:

> Note that Python datetime objects use your local time by default, whereas PyEphem expects time to be expressed in UTC. This means that passing datetime.now() to PyEphem will give unexpected results. Instead, use the datetime.utcnow() constructor:
> -- https://rhodesmill.org/pyephem/date#conversions

```python
>>> d = datetime.utcnow()
>>> print(ephem.Date(d))
2015/12/14 15:42:14
>>> d = datetime.fromtimestamp(1450107734, tz=timezone.utc)
>>> print(ephem.Date(d))
2015/12/14 15:42:14
```

https://github.com/pvlib/pvlib-python/blob/17aed3af0fb2213ccc2fd57a91c7ee699a162ad9/pvlib/solarposition.py#L648-L652

https://github.com/pvlib/pvlib-python/blob/17aed3af0fb2213ccc2fd57a91c7ee699a162ad9/pvlib/solarposition.py#L663-L664

https://github.com/pvlib/pvlib-python/blob/17aed3af0fb2213ccc2fd57a91c7ee699a162ad9/pvlib/solarposition.py#L676-L677

For `nrel` dni_extra, it is true that the function does not care about the day of year but it should refer to the year and the month correctly for an unknown `delta_t`:

https://github.com/pvlib/pvlib-python/blob/5d80876dbbd5b930337182a7e46ffe9701e2a095/pvlib/solarposition.py#L1007-L1011